### ISSUE TITLE
File#pread and File#pwrite missing

### ISSUE THREAD

**Issue by djberg96** [oracle/truffleruby#3718]
```
ruby -v
truffleruby 24.1.1, like ruby 3.2.4, Oracle GraalVM Native [arm64-darwin20]
```

Expected:

```ruby
fh = File.open("delete_this.txt", "w+")
fh.puts("The quick brown fox.")
fh.pread(9,0) # => "The quick"
fh.pwrite('Der', 0) # => 3
fh.pread(9,0) # => "Der quick"
fh.close
```
But got: undefined method errors for both 'pread' and 'pwrite'

**Comment by eregon** [comment]
Thanks for the report.
Do you know anything using these methods?

**Comment by djberg96** [comment]
> Thanks for the report. Do you know anything using these methods? (I'm not sure why one would use them over regular read/write, and to help prioritize)

I wrap them in my io-extra gem. :)

But, more seriously, from the Linux manpages:

> The pread() and pwrite() system calls are especially useful in multithreaded applications.  They allow multiple threads to perform I/O on the same file descriptor without being affected by changes to the file offset by other threads.

**Comment by andrykonchin** [comment]
Fixed in 49a4475480089a2e67df0b2129b9f2e4d44556ef.
